src/utils.ts

```diff
@@ -293,14 +294,44 @@ export function getEnsTextRecord(ens: string, record: string, network = '1') {
   return call(provider, ENS_RESOLVER_ABI, [address, 'text', [ensHash, record]]);
 }
 
-export async function getSpaceUri(id, network = '1') {
+export async function getSpaceUri(
+  id: string,
+  network = '1'
+): Promise<string | null> {
   try {
     return await getEnsTextRecord(id, 'snapshot', network);
   } catch (e) {
-    return false;
+    console.log(e);
+    return null;
   }
 }
 
+export async function getEnsOwner(
+  ens: string,
+  network = '1'
+): Promise<string | null> {
+  const provider = getProvider(network);
+  return provider.resolveName(ens);
```

Comment:
Hi @samuveth `resolveName` will return the owner of the domain right? or controller? 
For many domains the controller is already set to a different address, for example, https://app.ens.domains/name/gitcoindao.eth/details We want to check the controller instead of the owner maybe.?

Reply:
@ChaituVR by owner you mean Registrant right? The `resolveName` will always return the Controller of the domain.

Reply:
It returns the ETH address. (marked in red) we want to get controller (marked in blue) right?
<img width="1065" alt="Untitled" src="https://user-images.githubusercontent.com/15967809/218742616-4c9ac352-b63a-47d4-8f90-84f86c997070.png">

if there is no ETH address set (example ensdao.eth) it returns null

Reply:
Hm interesting. But this is how we have handled so far on the frontend. I couldn't find any way in their docs to get the controller specifically @ChaituVR

Reply:
@ChaituVR 

> if there is no ETH address set (example ensdao.eth) it returns null

This only seems to be the case for `lookupAddress`. https://docs.ethers.org/v5/single-page/#/v5/api/providers/provider/-%23-Provider-lookupAddress

But `resolveName` will always return an address if the name is owned by someone. https://docs.ethers.org/v5/single-page/#/v5/api/providers/provider/-%23-Provider-ResolveName

Reply:
The other solution would be to directly get it from the contract like this 

```js
// Create a new instance of the ENS registry
const registryAddress = '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e';
const ensRegistry = new ethers.Contract(registryAddress, ['function resolver(bytes32) view returns (address)', 'function owner(bytes32) view returns (address)'], provider);

// Get the resolver of the ENS address
const ensName = 'example.eth'; // Replace with your ENS name
const nodeHash = ethers.utils.namehash(ensName);
const resolverAddress = await ensRegistry.resolver(nodeHash);

// Get the controller of the ENS address from the resolver
const resolverAbi = ['function getController(address) view returns (address)'];
const resolver = new ethers.Contract(resolverAddress, resolverAbi, provider);
const controller = await resolver.getController(nodeHash);

```

Reply:
Not sure what's best but I feel getting it from resolver should be good I think, let's wait for @bonustrack :)